### PR TITLE
test: reliably generate incorrect signatures

### LINKED ISSUE #2941: Flickering test
https://app.circleci.com/pipelines/github/statechannels/statechannels/10584/workflows/d4814a01-de95-4df5-92d7-009e2dfb9b3e/jobs/52037

### PR DESCRIPTION
**Pull request by geoknee**
Fixes #2941

### CODE REVIEW

File: packages/nitro-protocol/test/contracts/ForceMove/challenge.test.ts [hunk 1]
```diff
@@ -180,7 +181,7 @@ describe('challenge', () => {
 
       switch (challengeSignatureType) {
         case 'incorrect':
-          challengeSignature = {...correctChallengeSignature, v: correctChallengeSignature.v + 1};
```
Comment (geoknee) [inline packages/nitro-protocol/test/contracts/ForceMove/challenge.test.ts:183]: This (very hacky) approach was sometimes generating a valid but incorrect signature (desired), and sometimes generating an entirely invalid signature (undesired)

File: packages/nitro-protocol/test/contracts/ForceMove/challenge.test.ts [hunk 2]
```diff
@@ -283,3 +284,23 @@ describe('forceMove with transaction generator', () => {
...
+  while (participants.find(participant => participant == nonParticipant)) {
+    const wallet = Wallet.createRandom();
+    nonParticipant = wallet.address;
+    pK = wallet.privateKey;
+  }
+  const challengeState = signedStates[signedStates.length - 1].state;
+  const challengeHash = hashChallengeMessage(challengeState);
+
+  return signData(challengeHash, pK);
+}
```
Comment (geoknee) [inline packages/nitro-protocol/test/contracts/ForceMove/challenge.test.ts:306]: This may be over-engineered, but it should be pretty reliable.

**Comment by andrewgordstewart** [comment]
Couldn't we used a constant, valid signature? The chance that it's correct is essentially zero

**Comment by geoknee** [comment]
> Couldn't we used a constant, valid signature? The chance that it's correct is essentially zero

2fb9c8f

File: packages/nitro-protocol/src/signatures.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ function hashMessage(hashedData: string): string {
   return utils.hashMessage(utils.arrayify(hashedData));
 }
 
-function signData(hashedData: string, privateKey: string): Signature {
+export function signData(hashedData: string, privateKey: string): Signature {
   const signingKey = new utils.SigningKey(privateKey);
 
   return utils.splitSignature(signingKey.signDigest(hashMessage(hashedData)));
```
Comment (snario) [inline packages/nitro-protocol/src/signatures.ts:77]: Why is the argument called `hashedData` when we pass it into `hashMessage`? I forget if we're intentionally double hashing, or if the naming is just confusing

Comment (geoknee) [reply packages/nitro-protocol/src/signatures.ts:77]: We are double hashing.

**Review by snario — APPROVED** [review]

**Pull request merged**